cql3/statements/alter_table_statement.cc

```diff
@@ -70,7 +70,15 @@ alter_table_statement::alter_table_statement(shared_ptr<cf_name> name,
 }
 
 future<> alter_table_statement::check_access(service::storage_proxy& proxy, const service::client_state& state) const {
-    return state.has_column_family_access(keyspace(), column_family(), auth::permission::ALTER);
+    return state.has_column_family_access(keyspace(), column_family(), auth::permission::ALTER).then([this, &state] {
+        const auto table = column_family();
+        if (!state.get_auth_service()->underlying_role_manager().safe(*this) ||
+            !state.get_auth_service()->underlying_authenticator().safe(*this) ||
```

Comment:
Why are role manager/authenticator consulted? Is it to check that you don't mess with the tables implementing them?

I think a better approach is to register their concerns with authorizer, so we don't have to hook every internal service that happens to have protected resources.

Reply:
> Why are role manager/authenticator consulted? Is it to check that you don't mess with the tables implementing them?

Yes.  This is equivalent to the old code here:
https://github.com/scylladb/scylla/blob/052a8d036d8c5ef73cb2ace333def29accde3634/auth/service.cc#L375-L377


> I think a better approach is to register their concerns with authorizer, so we don't have to hook every internal service that happens to have protected resources.

Why not hook them up to auth::service?  It's more generic and already plumbed through to permission checks.